test/integration/test_timeout.py

```diff
@@ -157,17 +154,16 @@ def test_timeout_cloudantv1_sessionauth(self):
             # Call the server
             my_service.get_server_information()
 
-            # Assert timeout is set to the authenticator
-            auth_args = Helpers.get_authenticate_arguments(requests.request)
-            Helpers.assert_default_timeout_setting(self, auth_args)
+            # first call is to a _session end-point, so we have an additional call on our mock request
+            if tc_num == 0:
+                # Assert timeout is set to the authenticator
+                auth_args = Helpers.get_request_arguments(self, my_service.authenticator.token_manager, tc_num, tc_num+2)
+                Helpers.assert_default_timeout_setting(self, auth_args)
 
             # Assert timeout is set in the server request
-            req_args = Helpers.get_request_arguments(self, my_service, tc_num)
+            req_args = Helpers.get_request_arguments(self, my_service, tc_num+1, tc_num+2)
             tc['assert_func'](self, req_args)
```

Comment:
It is confusing (and probably fragile to base this on the enumerated order) - if we need different assertions for each case I think we should check the specific test (e.g. `if set_timeout == Helpers.set_no_timeout`).
Although I don't fully understand why we need different assertions for 2 different timeout values. I think checking the auth timeout and the server request timeout should be valid for both test cases, but ofc they need to accommodate the session request first.